### PR TITLE
Test: try to test at pod level

### PR DESCRIPTION
**Pull request by SamYuan1990**
Signed-off-by: Sam Yuan <yy19902439@126.com>

### CODE REVIEW

**Comment by SamYuan1990** [comment]
@rootfs , @marceloamaral may I have a list of pod level metric we want to checking?

File: e2e/e2e_metric_test.go [hunk 1]
```diff
@@ -1,6 +1,3 @@
-//go:build bcc
```
Comment (rootfs) [inline e2e/e2e_metric_test.go:1]: keep this to avoid getting tested on mac

Comment (SamYuan1990) [reply e2e/e2e_metric_test.go:1]: I will fix this one later.
the major issue is about the test scope.
as matrix * pod * namespace?

**Review by marceloamaral — COMMENTED** [review]
The metrics in this test are a good start.

File: e2e/e2e_metric_test.go [hunk 2]
```diff
@@ -56,7 +57,12 @@ var _ = Describe("metrics check should pass", Ordered, func() {
 			switch et {
 			case textparse.EntrySeries:
 				m, _, v := p.Series()
+				p.Metric(&res)
 				k_metric[string(m)] = v
+				fmt.Println(string(m))
```
[the remark this answers is not in the document]
Comment (SamYuan1990) [reply e2e/e2e_metric_test.go:62]: yep, currently I print the output here just for discuss.
as metric * pod is a huge number, hence do we need check all of them or we just pick up key items?

[the remark this answers is not in the document]
Comment (rootfs) [reply e2e/e2e_metric_test.go:62]: please validate all the metrics, and if possible, record how long does the validation takes, so we'll have an idea about the scalability problem

File: e2e/e2e_metric_test.go (hunk 1 above)
Comment (rootfs) [reply e2e/e2e_metric_test.go:1]: yes, we need to validate for each pod, the kepler metrics are created and hopefully non-zero.

File: e2e/e2e_metric_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (SamYuan1990) [reply e2e/e2e_metric_test.go]: ok I will update the PR.

**Comment by SamYuan1990** [comment]
> https://github.com/sustainable-computing-io/kepler/pull/440#discussion_r1037489684
> how long does the validation takes, so we'll have an idea about the scalability problem

@rootfs , for end to end validation so far it nearly don't take too much time, with reasons:
- we get data from kepler restapi, and keep it in memory.
- we just have kubecore, monitoring and kepler namespaces... hence we don't have to much pods as a heavy workload.

**Comment by SamYuan1990** [comment]
@rootfs , @marceloamaral below metrics are zero with kind.
```
kepler_exporter_build_info
kepler_node_core_joules_total
kepler_node_dram_joules_total
kepler_node_energy_stat
kepler_node_nodeInfo
kepler_node_other_host_components_joules_total
kepler_node_package_energy_millijoule
kepler_node_package_joules_total
kepler_node_platform_joules_total
kepler_node_uncore_joules_total
kepler_container_gpu_joules_total
kepler_container_other_host_components_joules_total
kepler_container_uncore_joules_total
```

**Comment by rootfs** [comment]
@SamYuan1990 this looks good, can you do one more `go mod tidy`? The vendor file list looks quite long

**Comment by SamYuan1990** [comment]
@rootfs here you are.
```
yuanyi@yuanyideMacBook-Pro kepler % git status
On branch podcheckingine2e
nothing to commit, working tree clean
yuanyi@yuanyideMacBook-Pro kepler % go mod tidy
yuanyi@yuanyideMacBook-Pro kepler % git status
On branch podcheckingine2e
nothing to commit, working tree clean
```

**Review by rootfs — APPROVED** [review]

**Pull request merged**